### PR TITLE
feat(boards): Add BT60 Boards

### PR DESCRIPTION
**Pull request by ReFil**
## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [x] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] `.conf` file has optional extra features commented out

### CODE REVIEW

File: app/boards/arm/bt60/bt60.dtsi [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+* SPDX-License-Identifier: MIT
+*/
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+#include <dt-bindings/zmk/matrix_transform.h>
+
+/ {
+	model = "BT60";
+	compatible = "bt60";
```
Comment (Nicell) [inline app/boards/arm/bt60/bt60.dtsi:13]: The compatible property should be in the form `<manufacturer>,<model>`. Maybe `pw,bt60` or `polarityworks,bt60`?

File: app/boards/arm/bt60/bt60.dtsi [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+left_encoder: encoder_left {
+				compatible = "alps,ec11";
+				label = "LEFT_ENCODER";
+				a-gpios = <&gpio1 1 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+				b-gpios = <&gpio1 3 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+				resolution = <4>;
+				status = "okay";
+				};
+
```
Comment (Nicell) [inline app/boards/arm/bt60/bt60.dtsi:38]: Formatting is a bit haphazard here

File: app/boards/arm/bt60/bt60_v1.keymap [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		zmk,matrix_transform = &ansi_transform;
+	#elif defined(HHKB)
+		zmk,matrix_transform = &hhkb_transform;
+	#elif defined(ISO)
+		zmk,matrix_transform = &iso_transform;
+	#elif defined(ALL_1U)
+		zmk,matrix_transform = &all_1u_transform;
+	#else
+		zmk,matrix_transform = &split_transform;
+	#endif
```
Comment (Nicell) [inline app/boards/arm/bt60/bt60_v1.keymap:25]: Very neat! First time I've seen transform selection like this 👍

File: app/boards/arm/bt60/bt60_v1.zmk.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+name: BT60 V1 Soldered
+type: board
+arch: arm
+features:
+  - keys
+  - encoder
+outputs:
+  - usb
+  - ble
+url: https://polarityworks.com
```
Comment (Nicell) [inline app/boards/arm/bt60/bt60_v1.zmk.yml:12]: This URL doesn't go anywhere currently

File: app/boards/arm/bt60/bt60_v1_hs.zmk.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+name: BT60 V1 Hotswap
+type: board
+arch: arm
+features:
+  - keys
+  - encoder
+outputs:
+  - usb
+  - ble
+url: https://polarityworks.com
```
Comment (Nicell) [inline app/boards/arm/bt60/bt60_v1_hs.zmk.yml:12]: Same here on the URL

**Review by Nicell — COMMENTED** [review]
A few comments!

File: app/boards/arm/bt60/bt60_v1.keymap (hunk 3 above)
Comment (ReFil) [reply app/boards/arm/bt60/bt60_v1.keymap:25]: Not my idea tbh, Pete suggested it to me

File: app/boards/arm/bt60/bt60_v1.zmk.yml (hunk 4 above)
Comment (ReFil) [reply app/boards/arm/bt60/bt60_v1.zmk.yml:12]: Website is under development and will be up soon, we're working on it

Comment (Nicell) [reply app/boards/arm/bt60/bt60_v1.zmk.yml:12]: Do you guys have a more direct URL to the BT60? If not, could you update this in the future when you do? We can merge for now 👍

Comment (ReFil) [reply app/boards/arm/bt60/bt60_v1.zmk.yml:12]: our web dev hasn't given us the structure of the website yet unfortunately, sorry. I'll be sure to update it when they do

**Review by Nicell — APPROVED** [review]

**Pull request merged**